components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -395,12 +395,35 @@ func virtualServiceName(kfName string, namespace string) string {
 	return fmt.Sprintf("notebook-%s-%s", namespace, kfName)
 }
 
+// Function for the VirtualService header blacklist
+func Find(slice []string, val string) (int, bool) {
```

Comment:
Two comments here:
1. nit: Let's use lowercase function names for functions that we do not export (uppercase functions are exported and become part of the package api in Go)
2. The function is called `Find`, but you only use its second argument. I think what you want is a function that checks for existence, cause that's what you check. So how about renaming this to `exists` and returning only a bool?

Reply:
How do you want it to return only the bool?

Reply:
This is what I mean:
```go
func exists(slice []string, val string) bool {
	for i, item := range slice {
		if item == val {
			return true
		}
	}
	return false
}
```